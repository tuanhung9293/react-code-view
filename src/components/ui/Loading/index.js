import React from 'react'
import styles from './Loading.module.css'

function Loading() {
  return <div className={styles.spinner} />
}

export default Loading