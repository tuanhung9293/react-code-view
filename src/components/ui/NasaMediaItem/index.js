import React from 'react'
import MediaThumb from './MediaThumb'
import moment from 'moment'
import styles from './NasaMediaItem.module.css'
import clock from 'assets/clock.svg'

function NasaMediaItem({itemData, source, onFetchCollectionUrl, actionComponent: ActionComponent}) {
  return (
    <div className={styles.wrapper}>
      <div className={styles.inner}>
        <MediaThumb
          itemData={itemData}
          source={source}
          onFetchCollectionUrl={onFetchCollectionUrl}
        />
        <div className={styles.mediaType}>
          <a>{itemData.mediaType}</a>
        </div>
        <div className={styles.info}>
          <h4 className={styles.title}>
            <a>{itemData.title}</a>
          </h4>
          <div className={styles.meta}>
            <div className={styles.postauthor}>By <a>{itemData.photographer || 'unknown'}</a></div>
            <div>
              &nbsp;&nbsp;/&nbsp;&nbsp;<img src={clock} height="10px" />
              &nbsp;{moment(new Date(itemData.dateCreated)).format('Do MMMM YYYY')}
            </div>
          </div>
          <div className={styles.description}>{itemData.description}</div>
        </div>

        <div className={styles.addButton}>
          {ActionComponent && <ActionComponent itemData={itemData} />}
        </div>
      </div>
    </div>
  )
}

export default NasaMediaItem
