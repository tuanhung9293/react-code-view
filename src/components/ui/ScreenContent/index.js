import React from 'react'
import styles from './ScreenContent.module.css'

function ScreenContent(props) {
  return <div {...props} className={styles.base} />
}

export default ScreenContent