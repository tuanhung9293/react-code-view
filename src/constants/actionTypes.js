
export function apiActionType(type) {
  return {
    PENDING: `${type}_PENDING`,
    SUCCESS: `${type}_SUCCESS`,
    ERROR: `${type}_ERROR`
  }
}

export const START_TRACKING_WINDOW = 'START_TRACKING_WINDOW'
export const UPDATE_WINDOW_SIZE = 'UPDATE_WINDOW_SIZE'
export const STOP_TRACKING_WINDOW = 'STOP_TRACKING_WINDOW'

export const SET_INITIAL_FORM_VALUES = 'SET_INITIAL_FORM_VALUES'
export const SET_FORM_VALUE = 'SET_FORM_VALUE'
export const FORM_FIELD_BLUR = 'FORM_FIELD_BLUR'
export const CLEAR_FORM = 'CLEAR_FORM'
export const VALIDATE_FORM = 'VALIDATE_FORM'
export const VALIDATE_FIELD = 'VALIDATE_FIELD'
export const FORM_SUBMIT = apiActionType('FORM_SUBMIT')

export const SEARCH_NASA_MEDIA = apiActionType('SEARCH_NASA_MEDIA')
export const SEARCH_MORE_NASA_MEDIA = apiActionType('SEARCH_MORE_NASA_MEDIA')
export const FETCH_MEDIA_ITEM_COLLECTION = apiActionType('FETCH_MEDIA_ITEM_COLLECTION')

export const ADD_COLLECTION = 'ADD_COLLECTION'
export const EDIT_COLLECTION = 'EDIT_COLLECTION'
export const REMOVE_COLLECTION = 'REMOVE_COLLECTION'
export const LIKE_COLLECTION = 'LIKE_COLLECTION'
export const SORT_COLLECTION = 'SORT_COLLECTION'
