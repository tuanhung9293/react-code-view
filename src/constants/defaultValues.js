export const AUDIO_THUMB_URL_DEFAULT = 'AUDIO_THUMB_URL_DEFAULT'
export const PAGINATION_UNIT = 20

export const AVAILABLE_MEDIA_TYPES = [
  {id: 'image', value: 'image', message: 'Image'},
  {id: 'video', value: 'video', message: 'Video'},
  {id: 'audio', value: 'audio', message: 'Audio'}
]

export const AVAILABLE_SORT_TYPES = [
  {id: 'editDec', value: 'dateEdited,1', message: 'Default'},
  {id: 'titleDec', value: 'title,1', message: 'Title decrease'},
  {id: 'dateDec', value: 'dateCreated,1', message: 'Date decrease'},
  {id: 'titleInc', value: 'title,-1', message: 'Title increase'},
  {id: 'dateInc', value: 'dateCreated,-1', message: 'Date increase'}
]