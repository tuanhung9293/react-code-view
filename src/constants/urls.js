export const SEARCH_API_URL = 'https://images-api.nasa.gov'