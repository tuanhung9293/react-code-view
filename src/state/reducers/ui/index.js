import {UPDATE_WINDOW_SIZE} from 'constants/actionTypes'

function ui(state = {
  windowSize: {
    width: window.innerWidth,
    height: window.innerHeight
  }
}, action) {
  switch (action.type) {
    case UPDATE_WINDOW_SIZE:
      return {
        ...state,
        windowSize: {
          ...state.windowSize,
          ...action.payload
        }
      }
    default:
      return state
  }
}

export default ui