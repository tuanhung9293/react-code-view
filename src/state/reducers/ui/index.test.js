import ui from './'
import {UPDATE_WINDOW_SIZE} from 'constants/actionTypes'

describe('window size', () => {
  it('sets initial state', () => {
    expect(ui(undefined, {})).toMatchSnapshot()
  })

  it('updates window size', () => {
    expect(ui(undefined, {type: UPDATE_WINDOW_SIZE, payload: {
      width: 666,
      height: 666
    }})).toMatchSnapshot()
  })

  it('calculates new windowSize when undefined state is provided', () => {
    ui(undefined, {})
    window.innerHeight = 10000
    window.innerWidth = 10000
    expect(ui(undefined, {})).toMatchSnapshot()
  })
})
