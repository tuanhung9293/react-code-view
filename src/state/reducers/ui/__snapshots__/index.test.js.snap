// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`window size calculates new windowSize when undefined state is provided 1`] = `
Object {
  "windowSize": Object {
    "height": 10000,
    "width": 10000,
  },
}
`;

exports[`window size sets initial state 1`] = `
Object {
  "windowSize": Object {
    "height": 768,
    "width": 1024,
  },
}
`;

exports[`window size updates window size 1`] = `
Object {
  "windowSize": Object {
    "height": 666,
    "width": 666,
  },
}
`;
